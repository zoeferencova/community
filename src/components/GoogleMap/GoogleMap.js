import React, { useRef, useEffect } from "react";

export default function GoogleMaps(props) {
    const googleMapRef = React.createRef();
    const googleMap = useRef(null);
    const marker = useRef(null);
    const circle = useRef(null);
    const defaultRadius = 1609.344;
    const defaultLocation = { lat: 40.7450271, lng: -73.8858674 }
    const radius = parseFloat(props.radius)*1609.344 || defaultRadius;
    const location = props.location || defaultLocation;

    const mapStyles = {
        height: "100%",
        width: "100%",
    };

    const createGoogleMap = () => {
        return new window.google.smaps.Map(googleMapRef.current, {
            zoom: 14,
            center: location,
            disableDefaultUI: true,
        });
    }
        

    const createMarker = () => {
        props.displayMarker &&
        new window.google.maps.Marker({
            position: location,
            map: googleMap.current
        });
    }
        
    const createCircle = () => {
        return new window.google.maps.Circle({
            strokeColor: '#4d9e9a',
            strokeOpacity: 0.8,
            strokeWeight: 1,
            fillColor: '#84c4c1',
            fillOpacity: 0.35,
            map: googleMap.current,
            center: location,
            radius: radius
        });
    }
        

    useEffect(() => {
        googleMap.current = createGoogleMap();
        marker.current = createMarker();
        circle.current = createCircle();
        googleMap.current.fitBounds(circle.current.getBounds())
    });

    return (
        <div id="google-map" ref={googleMapRef} style={mapStyles} />
    )
}