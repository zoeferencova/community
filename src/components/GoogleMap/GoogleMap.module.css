.map {
    height: 300px !important;
}