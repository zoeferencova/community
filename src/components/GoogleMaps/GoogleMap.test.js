import React from "react";
import { shallow } from "enzyme";
import GoogleMap from "./GoogleMaps";

describe("GoogleMap", () => {
  it("renders GoogleMap component", () => {
    const wrapper = shallow(<GoogleMap />);

    expect(wrapper).toMatchSnapshot();
  })
})



