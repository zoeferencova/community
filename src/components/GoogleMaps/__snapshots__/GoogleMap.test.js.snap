// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GoogleMap renders GoogleMap component 1`] = `
<div
  id="google-map"
  style={
    Object {
      "height": "100%",
      "width": "100%",
    }
  }
/>
`;
